In [1]:
library(Sleuth3)
elmasu = case2201
Age = elmasu$Age
Matings = elmasu$Matings

fitllm = glm(Matings~Age, family=poisson)
summary(fitllm)


Call:
glm(formula = Matings ~ Age, family = poisson)

Deviance Residuals: 
     Min        1Q    Median        3Q       Max  
-2.80798  -0.86137  -0.08629   0.60087   2.17777  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept) -1.58201    0.54462  -2.905  0.00368 ** 
Age          0.06869    0.01375   4.997 5.81e-07 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 75.372  on 40  degrees of freedom
Residual deviance: 51.012  on 39  degrees of freedom
AIC: 156.46

Number of Fisher Scoring iterations: 5


In [3]:
fitho = glm(Matings~Age + I(Age^2), family=poisson)
summary(fitho)


Call:
glm(formula = Matings ~ Age + I(Age^2), family = poisson)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.8470  -0.8848  -0.1122   0.6580   2.1134  

Coefficients:
              Estimate Std. Error z value Pr(>|z|)
(Intercept) -2.8574060  3.0356383  -0.941    0.347
Age          0.1359544  0.1580095   0.860    0.390
I(Age^2)    -0.0008595  0.0020124  -0.427    0.669

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 75.372  on 40  degrees of freedom
Residual deviance: 50.826  on 38  degrees of freedom
AIC: 158.27

Number of Fisher Scoring iterations: 5


1. Consider the elephant mating example from lecture.

 1. _Both the binomial and the Poisson distributions provide probability models for counts. Is the binomial distribution appropriate for the number of successful matings of the male African elephants?_  
No, the binomial distribution has a upper bound for the number of success. (Number of success in a definite number of trials). 
 2. _For the model fit in lecture, interpret the coefficient of age._  
 Keeping other explanatory variables constant, one year increase in age will increase the number of matings by $e^{\beta_1}-1 = e^{0.0687}-1=0.07$
 3. _Consider the plot of the number of matings versus age. The spread of the responses is larger for larger values of the mean response. Should we be concerned?_  
No, the model assumes the underlying distribution is Poisson, where $E(X)=var(X)=\mu$. Therefore, the spread will increase as the increase of of the mean response 
 4. _From the estimated log-linear regression of the elephants' successful matings on age, what are the mean and variance of counts of successful matings (in the 8 years of the study) for the elephants who are aged 25 years at the beginning of the observation period? What are the mean and variance for elephants who are aged 45 years?_
 For 25 years old: mean = variance = $e^{-1.582+(0.0687)25}=1.15$
 For 25 years old: mean = variance = $e^{-1.582+(0.0687)45}=4.53$
 5. _While it is hypothesized that the number of matings increases with age, there
may be an optimal age for matings where, for older elephants the number of
matings starts to decline. One way to investigate this is to add a quadratic term
for age into the model to allow the log of the mean number of matings to reach
a peak. Does the inclusion in the model of $age^2$ improve the fit?_   
By Wald test, since $p=0.669$, we cannot reject the null hypothesis, there is no evidence that the higher order term improve the fit

2. What is the difference between a log-linear model and a linear model after the log transformation of the response?

For log-linear model, mean is $\mu$, the model is $\log(\mu)=X\beta$.

For SLR with transformation, the model is expressed in terms of the mean of the log of $Y$ with different model assumptions. 
 - different variance
 - different test procedures and diagnosis
 


3. Why are ordinary residuals $y_i-\hat\mu_i$ not particularly useful for Poisson regression?

Because the variance in Poisson model is not constant. The residuals with larger means will have larger variances. So if an observation has
a large residual it is diffcult to know whether it is an outlier or an observation from
a distribution with larger variance than the others. Residuals that are studentized
so that they have the same variance are more useful for identifying outliers.

4. Consider the deviance goodness-of-t test.
 1. _Under what conditions is it valid for Poisson regression?_  
 The sample size for each group is large enough. 
 2. _When it is valid, what possibilities are suggested by a small p-value?_
 The Poisson distribution is an inadequate model, the explanatory variables are inadequate, or there are some outliers. 
 3. _Large p-value?_  
 Model is correct, or there is insufficient data to detect any inadequacies. 

5. Because the underlying distribution is Poission. The likelihood function is
$$\prod_1^n \frac{\mu_i^{y_i}e^{-\mu_i}}{y_i!}=\frac{e^{-\sum \mu_i}\prod \mu_i^{y_i}}{\prod y_i!}$$
Take log
$$=-\sum \mu_i + \sum y_i \log(mu_i) - \sum \log(y_i!)$$
Where $\mu_i = e^{X_{\cdot,i}\beta}$ (X: the ith observation)
